src/acceptanceTest/java/uk/gov/hmcts/futurehearings/hmi/acceptance/listings/GETListingsValidationTest.java

```diff
@@ -0,0 +1,118 @@
+        this.setHttpSucessStatus(HttpStatus.OK);
+        this.setRelativeURLForNotFound(this.getRelativeURL().replace("listings","listing"));
+        this.setHmiSuccessVerifier(new GETListingsValidationVerifier());
+        this.setHmiErrorVerifier(new HMICommonErrorVerifier());
+    }
+
+    @ParameterizedTest(name = "Listing Case Id with and without values - Param : {0} --> {1}")
+    @CsvSource(value = {"listing_id_casehq, 1234", "listing_id_casehq,' '", "listing_id_casehq,NIL"}, nullValues= "NIL")
+    void test_listing_id_queryparam_with_value(final String sessionStartDateHQKey, final String sessionStartDateValue) throws IOException {
+        this.setUrlParams(buildQueryParams(sessionStartDateHQKey, sessionStartDateValue));
+        commonDelegate.test_expected_response_for_supplied_header(getApiSubscriptionKey(),
+                getRelativeURL(), getInputPayloadFileName(),
+                createStandardPayloadHeader(getApiSubscriptionKey()),
+                null,
+                getUrlParams(),
+                getHttpMethod(),
+                HttpStatus.OK, getInputFileDirectory(),
+                getHmiSuccessVerifier(),
+                LISTINGS_SUCCESS_MSG);
+    }
+
+    @ParameterizedTest(name = "Date of listing with and without values - Param : {0} --> {1}")
+    @CsvSource(value = {"date_of_listing, 2018-01-29 21:36:01Z", "date_of_listing,' '", "date_of_listing,NIL"}, nullValues= "NIL")
+    void test_date_of_listing_queryparam_with_value(final String dateOfListingKey, final String dateOfListingValue) throws IOException {
+        this.setUrlParams(buildQueryParams(dateOfListingKey, dateOfListingValue));
+        commonDelegate.test_expected_response_for_supplied_header(getApiSubscriptionKey(),
+                getRelativeURL(), getInputPayloadFileName(),
+                createStandardPayloadHeader(getApiSubscriptionKey()),
+                null,
+                getUrlParams(),
+                getHttpMethod(),
+                HttpStatus.OK, getInputFileDirectory(),
+                getHmiSuccessVerifier(),
+                LISTINGS_SUCCESS_MSG);
+    }
+
+    @ParameterizedTest(name = "Hearing Type with and without values - Param : {0} --> {1}")
+    @CsvSource(value = {"hearing_type, VH", "hearing_type,' '", "hearing_type,NIL"}, nullValues= "NIL")
+    void test_hearing_type_queryparam_with_value(final String hearingTypeKey, final String hearingTypeValue) throws IOException {
+        this.setUrlParams(buildQueryParams(hearingTypeKey, hearingTypeValue));
+        commonDelegate.test_expected_response_for_supplied_header(getApiSubscriptionKey(),
+                getRelativeURL(), getInputPayloadFileName(),
+                createStandardPayloadHeader(getApiSubscriptionKey()),
+                null,
+                getUrlParams(),
+                getHttpMethod(),
+                HttpStatus.OK, getInputFileDirectory(),
+                getHmiSuccessVerifier(),
+                LISTINGS_SUCCESS_MSG);
+    }
+
```

Comment:
One More parametarised test please that takes in 2 query parameters at a time with valid and invalid values...

Will discuss....

Reply:
Tests refactored and added to the code base as per the commit below...